crates/flux-fhir-analysis/src/conv/mod.rs

```diff
@@ -1158,6 +1162,21 @@ impl<'genv, 'tcx: 'genv, P: ConvPhase<'genv, 'tcx>> ConvCtxt<P> {
             .find_by_name_and_kind(self.tcx(), assoc_name, assoc_kind, trait_def_id)
     }
 
+    fn normalize_sorts(&self, sort: rty::Sort) -> rty::Sort {
+        let genv = self.genv();
+        let infcx = genv
+            .tcx()
+            .infer_ctxt()
+            .build(TypingMode::non_body_analysis());
+        if let Some(def_id) = self.owner().def_id()
+            && let Ok(normalized_sort) = sort.normalize_sorts(def_id.into(), genv, &infcx)
+        {
+            normalized_sort
+        } else {
+            sort
+        }
+    }
```

Comment:
This is the function I was talking about. We should normalize there. Note that `normalize_sorts` already normalizes weak aliases.

https://github.com/flux-rs/flux/blob/main/crates/flux-fhir-analysis/src/wf/sortck.rs#L513

Reply:
Ok I moved the call to `normalize_sort` to this place, please take a look!